### DISCUSSION TITLE
spring boot kotlin/webflux + reactive datastore using mongodb and r2dbc

### DISCUSSION THREAD

**Discussion by geromegrignon** [realworld-apps/realworld#995] [Added implementations]
* Spring webflux kotlin is a lot more maintainable than the java verion. 
* Spring data works with reactive mongo db, and reactive wrapper of a relational database (r2dbc, h2 in this case) together. This allows a small project use a persistence tier technologies of the best fit. For example, use rdbms for OLTP, and use nosql for the transaction ledge.
* This project also show case a way to separate from the detail of the persistence tier that allows delayed decision on which persistence technology will be used for production. This project used reactive mongodb for the user management feature. A data service layer is added to separate the service from the repository and made it easier to switch to use r2dbc (h2), or other reactive persistence technologies in the future.

**Comment by geromegrignon** [comment]
@alphawyu Hi I moved your discussion here as it was added in the 'work in progress' section.

**Reply by alphawyu** [reply]
that was my mistake to left out the subject. 
thank you for the update.

**Reply by alphawyu** [reply]
the project url is https://github.com/alphawyu/spring-webflux-mongodb-r2dbc
